docs(typescript): update sdk version in typescript definition header

ewanharris opened the pull request:
Not sure whether this is what you had in mind but it looks like the docgen script only reads in the version for generating changes, figured it made sense to just require in the package.json in the write header func.

ewanharris committed:
docs(typescript): update sdk version in typescript definition header

build commented:
<!--
  0 failure: 
  2 warning:  There is no linke..., :mag: Can't find ...
  
  
  DangerID: danger-id-default;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a></td>
    </tr>
  
<tr>
      <td>:warning:</td>
      <td>

  :mag: Can't find junit reports at `./junit.*.xml`, skipping generating JUnit Report.
  </td>
    </tr>
  </tbody>
</table>



<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against d457301bf610879a596eb745bbfc9402cecd3b06
</p>

ewanharris commented:
Validating locally using the linting in DefinitelyTyped throws some errors, will fix those in this PR

ewanharris committed:
fix: ensure generated typescript def file passing linting

Correct a couple of invalid docs, remove double newline from end of file

ewanharris committed:
docs(typescript): mark as non-npm package

Done by DefinitelyTyped maintainer in https://github.com/DefinitelyTyped/DefinitelyTyped/commit/608c146d8982a2ff7307f0a3bdddd6d55cb1a63e#diff-948841420dd57be7be5d9d21fe002f4c so we should keep this

janvennemann commented:
@ewanharris Are the changes to the .yml files requires to pass linting or why are they in this PR?

ewanharris commented:
@janvennemann, yes they are required to pass the linting step in DefinitelyTyped.

It was throwing the following errors
* `propertyList` didn't have typed array
* `description/pin/title` didn't have a jsdoc description when generated, the docgen typescript template pulls from summary which [per the tdoc spec is required](https://wiki.appcelerator.org/display/guides2/TDoc+Specification#TDocSpecification-PropertySpecification) and description is optional so I switched description to summary for those
* Double new line at the end of the file which is the change on line 384

build commented:
<!--
  0 failure: 
  1 warning:  There is no linke...
  2 messages
  
  DangerID: danger-id-Danger;
-->


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Warnings</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:warning:</td>
      <td>There is no linked JIRA ticket in the PR body. Please include the URL of the relevant JIRA ticket. If you need to, you may file a ticket on <a href="https://jira.appcelerator.org/secure/CreateIssue!default.jspa">JIRA</a></td>
    </tr>
  </tbody>
</table>


<table>
  <thead>
    <tr>
      <th width="50"></th>
      <th width="100%" data-danger-table="true">Messages</th>
    </tr>
  </thead>
  <tbody><tr>
      <td>:book:</td>
      <td>

  :floppy_disk: <a href="https://jenkins.appcelerator.org/job/titanium-sdk/job/titanium_mobile/job/PR-10822/6/artifact/dist/mobilesdk-8.1.0.v20190423100522-osx.zip">Here's the generated SDK zipfile</a>.
  </td>
    </tr>
  
<tr>
      <td>:book:</td>
      <td>

  :white_check_mark: All tests are passing
Nice one! All 3793 tests are passing.
(There are 466 tests skipped)
  </td>
    </tr>
  </tbody>
</table>


<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 5784903d236ad59267721c1f11bb1b16f2377425
</p>
